modules/host/host.go

```diff
@@ -148,6 +158,38 @@ func (h *Host) Capacity() int64 {
 	return h.spaceRemaining
 }
 
+// Close shuts down the host, preparing it for garbage collection.
+func (h *Host) Close() error {
+	// The order in which things are closed has been explicitly chosen to
+	// minimize turbulence in the event of an error.
+
+	// Save the latest host state.
+	err := func() error {
+		h.mu.Lock()
+		defer h.mu.Unlock()
+		return h.save()
+	}()
```

Comment:
I really don't see why you need this pattern for a single line of under-lock code. It doesn't seem worth the trouble. I think it's saner to reserve the inline-`defer` pattern for large blocks of under-lock code where multiple calls to `Unlock` would be required, and to stick a comment above each of them so that the reader isn't baffled by a random inline function.